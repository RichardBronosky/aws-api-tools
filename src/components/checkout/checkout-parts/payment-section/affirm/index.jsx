import React from 'react'
import AffirmButton from './affirm-button'
import { getAffirmBody } from '../../../../../lib/helpers/checkout/payment-section/affirm'

export default class AffirmButtonWrapper extends React.Component {
  state = {
    affirmLoaded: false,
  }

  affirmFunc = (l, g, m, e, a, f, b) => {
    var d,
      c = l[m] || {},
      h = document.createElement(f),
      n = document.getElementsByTagName(f)[0],
      k = function(a, b, c) {
        return function() {
          a[b]._.push([c, arguments])
        }
      }
    c[e] = k(c, e, 'set')
    d = c[e]
    c[a] = {}
    c[a]._ = []
    d._ = []
    c[a][b] = k(c, a, b)
    a = 0
    for (b = 'set add save post open empty reset on off trigger ready setProduct'.split(' '); a < b.length; a++)
      d[b[a]] = k(c, e, b[a])
    a = 0
    for (b = ['get', 'token', 'url', 'items']; a < b.length; a++) d[b[a]] = function() {}
    h.async = !0
    h.src = g[f]
    n.parentNode.insertBefore(h, n)
    delete g[f]
    d(g)
    l[m] = c
    this.setState({ affirmLoaded: true })
  }

  affirmClick = () => {
    affirm.checkout(getAffirmBody())
    affirm.checkout.open()
  }

  render() {
    const _affirm_config = {
      public_api_key: process.env.GATSBY_AFFIRM_API_KEY,
      script: `${ process.env.GATSBY_AFFIRM_URL }/js/v2/affirm.js`,
    }
    return (
      <AffirmButton
        affirmLoaded={ this.state.affirmLoaded }
        affirmFunc={ this.affirmFunc }
        affirm_config={ _affirm_config }
        affirmClick={ this.affirmClick }
      />
    )
  }
}
