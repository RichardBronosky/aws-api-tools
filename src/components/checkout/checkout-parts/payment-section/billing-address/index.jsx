import React from 'react'
import BillingAddress from './billing-address'
import {
  validateBillingAddress,
  submitBillingAddress,
} from '../../../../../lib/helpers/checkout/payment-section/billing-address'

export default class BillingAddressWrapper extends React.Component {
  state = {
    invalidFields: [],
    loading: false,
  }

  submitBillingAddress = async () => {
    this.setBillingState({ loading: true })
    const invalidFields = await validateBillingAddress(this.props.order)
    this.setState({ invalidFields })
    if (invalidFields.length < 1) {
      submitBillingAddress(this.setBillingState)
    } else {
      this.setBillingState({ loading: false })
    }
  }

  setBillingState = state => this.setState(state)

  clearInvalidFields = () => this.setState({ invalidFields: [] })

  render() {
    const { order } = this.props
    const { invalidFields, loading } = this.state
    return (
      <BillingAddress
        order={ order }
        invalidFields={ invalidFields }
        loading={ loading }
        submitBillingAddress={ this.submitBillingAddress }
        clearInvalidFields={ this.clearInvalidFields }
        setBillingState={ this.setBillingState }
      />
    )
  }
}
