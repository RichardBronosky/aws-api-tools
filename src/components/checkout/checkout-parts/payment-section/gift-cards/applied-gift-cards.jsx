import React from 'react'
import { currencyFormatUS } from '../../../../../lib/helpers/string-helper'
import { onAddAnotherCard, onRemoveGiftCard } from '../../../../../lib/helpers/checkout/payment-section/gift-cards'
import loaderLight from '../../../../../assets/images/loader-light.svg'

export default ({ giftCards, order, setGiftCardState, removing }) => (
  <div className="grid-x">
    <div className="applied-gift-cards small-12">
      <h4>Applied Gift Cards</h4>
      <div className="grid-x">
        { giftCards.map(giftCard => (
          <div key={ giftCard.giftCardNumber } className="small-12 medium-7 large-6 applied-gift-card-container">
            <div className="applied-gift-card card">
              <button
                className="remove-card"
                tabIndex="0"
                value="Remove Gift Card"
                aria-label="Remove Gift Card"
                onClick={ () => onRemoveGiftCard(order, giftCard, setGiftCardState) }
              >
                { !removing && (
                  <img
                    className="icon close"
                    alt="close icon"
                    src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23FFFFFF' d='M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z' /%3E%3C/svg%3E"
                  />
                ) }
                { removing && <img className="loader" alt="Removing gift card" src={ loaderLight } /> }
              </button>
              <div className="gift-card-info">
                <p className="bold left">Gift Card</p>
                <p className="number">- { giftCard.giftCardNumber }</p>
                <br />
                Will use
                <p className="bold">
                  { currencyFormatUS(giftCard.authorizedAmount ? giftCard.authorizedAmount : giftCard.balance) }
                </p>
                of your <br />
                <p className="bold left">{ currencyFormatUS(giftCard.balance) }</p>
                balance
              </div>
            </div>
          </div>
        )) }
      </div>
      { !order.giftCardInfo.addAnotherCard && order && order.amountDue > 0 && (
        <button
          className="add-another-card"
          tabIndex="0"
          value="Remove Gift Card"
          aria-label="Remove Gift Card"
          onClick={ () => onAddAnotherCard(order) }
        >
          Add Another Gift Card
        </button>
      ) }
    </div>
  </div>
)
