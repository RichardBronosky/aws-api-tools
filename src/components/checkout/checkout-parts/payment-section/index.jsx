import React from 'react'
import { connect } from 'react-redux'
import { handleAffirm } from '../../../../lib/helpers/checkout/payment-section/affirm'
import PaymentSection from './payment-section'
import '../../../../assets/css/components/checkout/checkout-parts/payment-section/payment-section.sass'

class PaymentSectionWrapper extends React.Component {
  state = {
    morePaymentMethods: false,
    ccForm: null,
    mountLoad: false,
  }

  componentDidMount() {
    handleAffirm()
  }

  render() {
    const { paymentInvalidFields } = this.props
    return <PaymentSection { ...this.props } invalidFields={ paymentInvalidFields } />
  }
}

const mapStateToProps = state => {
  return {
    order: state.checkout.order,
    checkoutStep: state.checkout.checkoutStep,
    checkoutStepsCompleted: state.checkout.checkoutStepsCompleted,
    paymentInvalidFields: state.checkout.paymentInvalidFields,
  }
}

export default connect(mapStateToProps)(PaymentSectionWrapper)
