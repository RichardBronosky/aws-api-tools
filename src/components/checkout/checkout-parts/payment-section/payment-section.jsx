import React from 'react'
import classNames from 'classnames'
import ReactToolTip from 'react-tooltip'
import { setOrderInfo } from '../../../../lib/helpers/checkout/global'
import { shouldShowPayments } from '../../../../lib/helpers/checkout/payment-section/payment-section'
import { affirmAllowed as affirmIsAllowed } from '../../../../lib/helpers/checkout/payment-section/affirm'
import '../../../../assets/css/components/checkout/checkout-parts/payment-section/payment-section.sass'

import BillingAddress from './billing-address'
import PaymentCondensed from './payment-condensed'
import GiftCards from './gift-cards'
import CreditCardMicroform from './credit-card-microform'
import RoomsToGoCredit from './rooms-to-go-credit'
import ErrorMessage from '../checkout-error-message'
import { getCreditCardDecision } from '../../../../lib/helpers/checkout/payment-section/credit-card'

export default class PaymentSection extends React.Component {
  render() {
    const { order, checkoutStep, checkoutStepsCompleted, invalidFields } = this.props
    const showPaymentMethods = shouldShowPayments(order)
    const decision = getCreditCardDecision()
    const affirmAllowed = affirmIsAllowed(order)
    let hideMore = false
    if (order && order.paymentInfo && order.paymentInfo.length > 0) {
      const fin = order.paymentInfo.filter(payment => payment.paymentType === 'FIN')
      if (fin.length === 1 && order.paymentInfo.length === 1) {
        hideMore = order.amountDue <= 0
      }
    }
    const morePaymentMethods =
      order.selectedPaymentType === 'PayPal' ||
      order.selectedPaymentType === 'Visa Checkout' ||
      order.selectedPaymentType === 'Affirm' ||
      order.selectedPaymentType === 'ApplePay'
    return (
      <>
        { checkoutStep !== 'payment' && order.amountDue <= 0 && checkoutStepsCompleted.payment && (
          <PaymentCondensed order={ order } />
        ) }
        { checkoutStep === 'payment' && (
          <div className="payment-form">
            <p className="required-label">*Asterisks indicate required fields in the form below</p>
            { invalidFields.length > 0 && (
              <ErrorMessage customMessage="Cannot continue until a payment has been submitted." />
            ) }
            <div className="payment-form-container small-12 large-8">
              <div
                className={ classNames('gift-card-container', {
                  'gift-cards-active': order.giftCardInfo.useGiftCard,
                }) }
              >
                <GiftCards order={ order } />
              </div>
              { (order.amountDue > 0 || !order.amountDue) && showPaymentMethods && (
                <div className="grid-x" role="tablist" aria-label="Payment Method">
                  <button
                    className={ classNames('payment-type cell small-4', {
                      active: order.selectedPaymentType === 'Credit' && !morePaymentMethods,
                      disabled: hideMore,
                    }) }
                    role="tab"
                    id="tab1"
                    data-tip
                    data-tip-disable={ !hideMore }
                    data-for="creditDisabled"
                    tabIndex={ order.selectedPaymentType === 'Credit' && !morePaymentMethods ? 0 : -1 }
                    aria-selected={ order.selectedPaymentType === 'Credit' && !morePaymentMethods }
                    aria-controls="panel1"
                    onClick={ () => {
                      if ((order.selectedPaymentType !== 'Credit' || morePaymentMethods) && !hideMore) {
                        setOrderInfo('Credit', 'selectedPaymentType')
                      }
                    } }
                  >
                    <span className="payment-type-text">Credit Card</span>
                  </button>
                  <ReactToolTip id="creditDisabled" place="top" type="dark" effect="float">
                    Must remove financing to use another payment method.
                  </ReactToolTip>
                  <button
                    className={ classNames('payment-type cell small-4', {
                      active: order.selectedPaymentType === 'Rooms To Go' && !morePaymentMethods,
                    }) }
                    role="tab"
                    id="tab2"
                    tabIndex={ order.selectedPaymentType === 'Rooms To Go' && !morePaymentMethods ? 0 : -1 }
                    aria-selected={ order.selectedPaymentType === 'Rooms To Go' && !morePaymentMethods }
                    aria-controls="panel2"
                    onClick={ () => {
                      setOrderInfo('Rooms To Go', 'selectedPaymentType')
                      this.setState({
                        morePaymentMethods: false,
                      })
                    } }
                  >
                    <span className="payment-type-text">Rooms To Go Finance</span>
                  </button>
                  <button
                    className={ classNames('payment-type cell small-4', {
                      active: morePaymentMethods,
                      disabled: hideMore,
                    }) }
                    role="tab"
                    id="tab3"
                    tabIndex={ morePaymentMethods ? 0 : -1 }
                    aria-selected={ morePaymentMethods }
                    aria-controls="panel3"
                    data-tip="React-tooltip"
                    data-tip-disable={ !hideMore }
                    onClick={ () => {
                      if (!hideMore) {
                        if (order.paymentInfo.filter(payment => payment.paymentType === 'VISA').length) {
                          setOrderInfo('Visa Checkout', 'selectedPaymentType')
                        } else if (order.paymentInfo.filter(payment => payment.paymentType === 'AFF').length) {
                          setOrderInfo('Affirm', 'selectedPaymentType')
                        } else {
                          setOrderInfo('PayPal', 'selectedPaymentType')
                        }
                      }
                    } }
                  >
                    <span className="payment-type-text">Other Payment Options</span>
                  </button>
                  <ReactToolTip place="top" type="dark" effect="float">
                    Must remove financing to use another payment method.
                  </ReactToolTip>
                </div>
              ) }

              <div id="panel1" role="tabpanel" tabIndex="0" className="credit-card-container">
                { showPaymentMethods && (
                  <>
                    { order.selectedPaymentType === 'Credit' && !morePaymentMethods && (
                      <div
                        className={ classNames('billing-iframe', {
                          accepted: decision,
                        }) }
                      >
                        <BillingAddress order={ order } />
                        <CreditCardMicroform order={ order } />
                      </div>
                    ) }
                  </>
                ) }
              </div>

              <div id="panel2" role="tabpanel" tabIndex="0" className="rooms-to-go-credit-container">
                { showPaymentMethods && order.selectedPaymentType === 'Rooms To Go' && <RoomsToGoCredit order={ order } /> }
              </div>

              <div id="panel3" role="tabpanel" tabIndex="0" className="grid-x more-payments">
                { showPaymentMethods && morePaymentMethods && (
                  <>
                    <div
                      className={ classNames('more-payment-type payment-type-image small-6 medium-3 large-3', {
                        active: order.selectedPaymentType === 'PayPal',
                      }) }
                      onClick={ () => setOrderInfo('PayPal', 'selectedPaymentType') }
                    >
                      <img
                        src="https://www.paypalobjects.com/webstatic/en_US/i/buttons/PP_logo_h_100x26.png"
                        alt="PayPal Logo"
                      />
                      <div className="help-link">
                        <a
                          href="https://www.paypal.com/webapps/mpp/paypal-popup"
                          title="What is PayPal?"
                          target="_blank"
                        >
                          What is PayPal?
                        </a>
                      </div>
                    </div>
                    <div
                      className={ classNames('more-payment-type payment-type-image small-6 medium-3 large-3', {
                        active: order.selectedPaymentType === 'Visa Checkout',
                      }) }
                      onClick={ () => setOrderInfo('Visa Checkout', 'selectedPaymentType') }
                    >
                      <img
                        alt="Visa Checkout"
                        src="https://assets.secure.checkout.visa.com/VCO/images/acc_99x34_blu01.png"
                      />
                      <div className="help-link">
                        <a className="v-learn" title="Learn More">
                          Learn More
                        </a>
                      </div>
                    </div>
                    <div
                      className={ classNames('more-payment-type small-6 medium-3 large-3', {
                        active: order.selectedPaymentType === 'Affirm',
                        'affirm-disabled': !affirmAllowed,
                      }) }
                      aria-label={ !affirmAllowed ? 'Not allowed to use Affirm for financing down payment.' : 'Use Affirm' }
                      data-tip
                      data-tip-disable={ affirmAllowed }
                      data-for="Affirm"
                      onClick={ () => affirmAllowed && setOrderInfo('Affirm', 'selectedPaymentType') }
                    >
                      <img
                        className="affirm-logo"
                        alt="Affirm Logo"
                        src="data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA1NzEuNjQgMTY2LjA0Ij48ZGVmcz48c3R5bGU+LmNscy0xe2ZpbGw6IzJiYzJkZjt9PC9zdHlsZT48L2RlZnM+PHRpdGxlPmxvZ290eXBlX2JsdWU8L3RpdGxlPjxwYXRoIGNsYXNzPSJjbHMtMSIgZD0iTTI5OC4zOSwwQTE3LjU3LDE3LjU3LDAsMSwwLDMxNiwxNy41NywxNy41OSwxNy41OSwwLDAsMCwyOTguMzksMFoiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDAgMCkiLz48cmVjdCBjbGFzcz0iY2xzLTEiIHg9IjI4My4zIiB5PSI0Ni42OCIgd2lkdGg9IjI5Ljk5IiBoZWlnaHQ9IjExOS4zMSIvPjxwYXRoIGNsYXNzPSJjbHMtMSIgZD0iTTQwMy40Nyw0My42MWMtMTUsMC0zMi4yNSwxMC44LTM3LjkzLDI0LjM0VjQ2LjY4SDMzNy4wOVYxNjZoMzBWMTEwLjU5YzAtMjMuNDUsOS0zNi41NCwyOC42MS0zNi45MUw0MTIuNDQsNDQuM0E2NC4xNyw2NC4xNywwLDAsMCw0MDMuNDcsNDMuNjFaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgwIDApIi8+PHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNTI0LDQzLjY5Yy0xMi44NywwLTI0LjQxLDQuODQtMzIuNSwxMy42MmwtMC40Mi40NS0wLjQxLS40NWMtOC04Ljc4LTE5LjQ4LTEzLjYyLTMyLjM2LTEzLjYyLTI3LjU4LDAtNDcuNiwyMC4xMS00Ny42LDQ3LjgxVjE2NmgyOS41MlY5MC44NmMwLTExLjYzLDcuMS0xOS4xNCwxOC4wOC0xOS4xNHMxOC4wOSw3LjUxLDE4LjA5LDE5LjE0VjE2Nkg1MDZWOTAuODZjMC0xMS42Myw3LjEtMTkuMTQsMTguMDktMTkuMTRzMTguMDksNy41MSwxOC4wOSwxOS4xNFYxNjZoMjkuNTFWOTEuNUM1NzEuNjQsNjMuOCw1NTEuNjIsNDMuNjksNTI0LDQzLjY5WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAwKSIvPjxwYXRoIGNsYXNzPSJjbHMtMSIgZD0iTTI0NywzNS42OWMwLTMuOTQuNTMtOC42NSwzLjktMTEuMjQsMy42OS0yLjg3LDkuMDktMi4zNCwxMy40Ni0yLjEybDUuODItMjEuNzItMy4xMS0uMTZjLTEyLjUyLS42Ny0yNS42OS0xLjYtMzYuNjcsNS42Ni05LjMxLDYuMTUtMTMuNDUsMTYuNzQtMTMuNDUsMjcuNjF2MTNIMTgxLjE4di0xMWMwLTMuOTIuNTItOC41OCwzLjgyLTExLjIsMy42OS0yLjkzLDkuMTYtMi4zOCwxMy41My0yLjE2bDUuODItMjEuNzItMy4xMS0uMTZjLTEyLjYyLS42OC0yNS45Mi0xLjYtMzYuOTEsNS44Ni05LjEyLDYuMTgtMTMuMTYsMTYuNjgtMTMuMTYsMjcuNDJ2MTNIMTM3LjY1VjY4LjM3aDEzLjUzVjE2NmgzMFY2OC4zN2gzNS43NlYxNjZoMzBWNjguMzdoMjAuNzdWNDYuNjhIMjQ3di0xMVoiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDAgMCkiLz48cGF0aCBjbGFzcz0iY2xzLTEiIGQ9Ik0xMjIuNDMsMTY2bDAtMTA1LjM2YTE3LjMyLDE3LjMyLDAsMCwwLTE1LjMyLTE3Yy01LjczLS4zNy0xMS44MiwxLjczLTE1LjQxLDYuMzhMMCwxNjZIMjIuNmM5LDAsMTYuMTgtNC42OSwyMS42MS0xMS43TDk1LDkwLjA3VjE2NmgyNy40NVoiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDAgMCkiLz48L3N2Zz4="
                      />

                      <div className="help-link">
                        <a
                          title="What is Affirm?"
                          target="_blank"
                          href="https://www.affirm.com/help/get-started/what-is-affirm"
                        >
                          What is Affirm?
                        </a>
                      </div>
                    </div>
                    <ReactToolTip id="Affirm" place="top" type="dark" effect="float">
                      Can't use Affirm to submit a down payment.
                    </ReactToolTip>
                    { /* { isSafari && (
                      <div
                        className={ classNames('more-payment-type small-12 medium-3 large-3', {
                          active: order.selectedPaymentType === 'ApplePay',
                        }) }
                        onClick={ () => setOrderInfo('ApplyPay', 'selectedPaymentType') }
                      >
                        <p className="payment-type-text">ApplePay</p>
                      </div>
                    ) } */ }
                  </>
                ) }
              </div>
            </div>
          </div>
        ) }
      </>
    )
  }
}
