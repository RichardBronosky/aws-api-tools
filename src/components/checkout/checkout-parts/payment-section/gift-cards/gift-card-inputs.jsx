import React from 'react'
import classNames from 'classnames'
import { onApplyGiftCard, setOrderGiftCardInfo } from '../../../../../lib/helpers/checkout/payment-section/gift-cards'
import loaderLight from '../../../../../assets/images/loader-light.svg'

export default ({ unableToAddMessage, order, invalidFields, loading, setGiftCardState }) => (
  <div className="gift-card grid-x">
    <div className="small-12 medium-6">
      <label className="" htmlFor="giftCardNumber">
        { unableToAddMessage !== '' && <p className="error">*{ unableToAddMessage }</p> }
        <p>*Gift Card Number</p>
        <input
          type="text"
          className={ classNames('gift-card-number', {
            invalid: invalidFields.includes('giftCardNumber'),
          }) }
          placeholder="Gift Card Number"
          name="Gift Card Number"
          aria-label="Gift Card Number"
          value={ order.giftCardInfo.giftCardNumber }
          onChange={ e => setOrderGiftCardInfo(e.target.value, 'giftCardNumber') }
        />
      </label>
      <br />
      <label className="" htmlFor="giftCardPin">
        <p>*Pin</p>
        <input
          type="text"
          className={ classNames('gift-card-pin', {
            invalid: invalidFields.includes('giftCardPin'),
          }) }
          placeholder="Pin"
          name="Pin"
          aria-label="Pin"
          value={ order.giftCardInfo.giftCardPin }
          onChange={ e => setOrderGiftCardInfo(e.target.value, 'giftCardPin') }
        />
      </label>
      <br />
      <button
        className="apply-gift-card-btn small-12 medium-3"
        tabIndex="0"
        value="Apply Gift Card"
        aria-label="Apply Gift Card"
        onClick={ () => onApplyGiftCard(order, setGiftCardState) }
      >
        { !loading ? 'apply gift card' : <img className="loader" alt="Removing gift card" src={ loaderLight } /> }
      </button>
      { order.giftCardInfo.addAnotherCard && (
        <button
          className="apply-gift-card-btn cancel small-12 medium-1"
          tabIndex="0"
          value="Cancel"
          aria-label="Cancel"
          onClick={ () => setOrderGiftCardInfo(false, 'addAnotherCard') }
        >
          Cancel
        </button>
      ) }
    </div>
  </div>
)
