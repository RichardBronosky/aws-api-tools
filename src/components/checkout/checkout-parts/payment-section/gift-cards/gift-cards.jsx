import React from 'react'
import CheckoutErrorMessage from '../../checkout-error-message'
import GiftCardInputs from './gift-card-inputs'
import AppliedGiftCards from './applied-gift-cards'
import { onChangeUseGiftCard } from '../../../../../lib/helpers/checkout/payment-section/gift-cards'
import '../../../../../assets/css/components/checkout/checkout-parts/payment-section/gift-cards.sass'

export default ({
  order,
  removing,
  loading,
  invalidFields,
  giftCards,
  paymentProperties,
  setGiftCardState,
  unableToAddMessage,
}) => (
  <>
    { order && (
      <>
        <input
          type="checkbox"
          name="Use Gift Card"
          aria-label="Use Gift Card"
          checked={ order.giftCardInfo.useGiftCard }
          onChange={ e => onChangeUseGiftCard(order, e.target.checked) }
        />
        Use gift card
        <br />
        { order.giftCardInfo.useGiftCard && paymentProperties && (
          <p className="note">
            Note: Adding or removing gift cards may require you to re-enter other payment information below.
          </p>
        ) }
        { order.giftCardInfo.useGiftCard && invalidFields.length > 0 && (
          <>
            { ' ' }
            <CheckoutErrorMessage invalidFields={ invalidFields } />
          </>
        ) }
        { giftCards.length > 0 && order.giftCardInfo.useGiftCard && (
          <AppliedGiftCards
            order={ order }
            removing={ removing }
            giftCards={ giftCards }
            setGiftCardState={ setGiftCardState }
          />
        ) }
        { ((giftCards && giftCards.length < 1) || order.giftCardInfo.addAnotherCard) && (
          <>
            { order.giftCardInfo.useGiftCard && (
              <GiftCardInputs
                unableToAddMessage={ unableToAddMessage }
                order={ order }
                setGiftCardState={ setGiftCardState }
                invalidFields={ invalidFields }
                loading={ loading }
              />
            ) }
          </>
        ) }
      </>
    ) }
  </>
)
