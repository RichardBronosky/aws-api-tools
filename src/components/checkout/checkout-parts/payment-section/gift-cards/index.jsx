import React from 'react'
import GiftCards from './gift-cards'
import { getGiftCards } from '../../../../../lib/helpers/checkout/payment-section/gift-cards'

export default class GiftCardsWrapper extends React.Component {
  state = {
    unableToAddMessage: '',
    loading: false,
    invalidFields: [],
    removing: false,
  }

  setGiftCardState = (info, field, callback = null) => {
    if (field) {
      this.setState(
        {
          ...this.state,
          [field]: info,
        },
        callback
      )
    } else {
      this.setState(
        {
          ...this.state,
          ...info,
        },
        callback
      )
    }
  }

  render() {
    const { order } = this.props
    const { removing, loading, invalidFields, unableToAddMessage } = this.state
    let paymentProperties = false
    if (order && order.paymentInfo && order.paymentInfo.length > 0) {
      paymentProperties = true
    }
    const giftCards = getGiftCards()
    return (
      <GiftCards
        order={ order }
        removing={ removing }
        loading={ loading }
        invalidFields={ invalidFields }
        giftCards={ giftCards }
        paymentProperties={ paymentProperties }
        setGiftCardState={ this.setGiftCardState }
        unableToAddMessage={ unableToAddMessage }
      />
    )
  }
}
