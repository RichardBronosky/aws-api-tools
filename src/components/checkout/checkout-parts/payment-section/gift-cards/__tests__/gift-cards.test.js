import React from 'react'
import renderer from 'react-test-renderer'
import GiftCards from '../gift-cards'
import * as giftCardHelper from '../../../../../../lib/helpers/checkout/payment-section/gift-cards'
import { order1, giftCards1 } from '../../../../../../lib/mocks/checkoutDataMocks'

describe('GiftCards', () => {
  let testProps = {
    order: {
      ...order1,
      giftCardInfo: {
        ...order1.giftCardInfo,
        useGiftCard: true,
      },
    },
    removing: false,
    loading: false,
    invalidFields: [],
    giftCards: giftCards1,
    paymentProperties: true,
    setGiftCardState: jest.fn(),
    unableToAddMessage: '',
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<GiftCards { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when not using gift card', () => {
    const tree = renderer
      .create(
        <GiftCards { ...{ ...testProps, order: { giftCardInfo: { ...order1.giftCardInfo, useGiftCard: false } } } } />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with invalid fields', () => {
    const tree = renderer.create(<GiftCards { ...{ ...testProps, invalidFields: ['test field'] } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly no other gift cards', () => {
    const tree = renderer.create(<GiftCards { ...{ ...testProps, giftCards: [] } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly ability to add another card', () => {
    const tree = renderer
      .create(
        <GiftCards { ...{ ...testProps, order: { giftCardInfo: { ...order1.giftCardInfo, addAnotherCard: true } } } } />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls onChangeUseGiftCard when simulating clicking use gift card checkbox', () => {
    giftCardHelper.onChangeUseGiftCard = jest.fn()
    const tree = renderer.create(<GiftCards { ...testProps } />)
    tree.root.findByProps({ name: 'Use Gift Card' }).props.onChange({ target: { checked: false } })
    expect(giftCardHelper.onChangeUseGiftCard).toHaveBeenCalled()
  })
})
