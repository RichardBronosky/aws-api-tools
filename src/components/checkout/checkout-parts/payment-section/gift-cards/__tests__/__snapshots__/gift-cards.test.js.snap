// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftCards renders correctly ability to add another card 1`] = `
Array [
  <input
    aria-label="Use Gift Card"
    checked={false}
    name="Use Gift Card"
    onChange={[Function]}
    type="checkbox"
  />,
  "Use gift card",
  <br />,
]
`;

exports[`GiftCards renders correctly no other gift cards 1`] = `
Array [
  <input
    aria-label="Use Gift Card"
    checked={true}
    name="Use Gift Card"
    onChange={[Function]}
    type="checkbox"
  />,
  "Use gift card",
  <br />,
  <p
    className="note"
  >
    Note: Adding or removing gift cards may require you to re-enter other payment information below.
  </p>,
  <div
    className="gift-card grid-x"
  >
    <div
      className="small-12 medium-6"
    >
      <label
        className=""
        htmlFor="giftCardNumber"
      >
        <p>
          *Gift Card Number
        </p>
        <input
          aria-label="Gift Card Number"
          className="gift-card-number"
          name="Gift Card Number"
          onChange={[Function]}
          placeholder="Gift Card Number"
          type="text"
          value=""
        />
      </label>
      <br />
      <label
        className=""
        htmlFor="giftCardPin"
      >
        <p>
          *Pin
        </p>
        <input
          aria-label="Pin"
          className="gift-card-pin"
          name="Pin"
          onChange={[Function]}
          placeholder="Pin"
          type="text"
          value=""
        />
      </label>
      <br />
      <button
        aria-label="Apply Gift Card"
        className="apply-gift-card-btn small-12 medium-3"
        onClick={[Function]}
        tabIndex="0"
        value="Apply Gift Card"
      >
        apply gift card
      </button>
    </div>
  </div>,
]
`;

exports[`GiftCards renders correctly when not using gift card 1`] = `
Array [
  <input
    aria-label="Use Gift Card"
    checked={false}
    name="Use Gift Card"
    onChange={[Function]}
    type="checkbox"
  />,
  "Use gift card",
  <br />,
]
`;

exports[`GiftCards renders correctly with default props 1`] = `
Array [
  <input
    aria-label="Use Gift Card"
    checked={true}
    name="Use Gift Card"
    onChange={[Function]}
    type="checkbox"
  />,
  "Use gift card",
  <br />,
  <p
    className="note"
  >
    Note: Adding or removing gift cards may require you to re-enter other payment information below.
  </p>,
  <div
    className="grid-x"
  >
    <div
      className="applied-gift-cards small-12"
    >
      <h4>
        Applied Gift Cards
      </h4>
      <div
        className="grid-x"
      >
        <div
          className="small-12 medium-7 large-6 applied-gift-card-container"
        >
          <div
            className="applied-gift-card card"
          >
            <button
              aria-label="Remove Gift Card"
              className="remove-card"
              onClick={[Function]}
              tabIndex="0"
              value="Remove Gift Card"
            >
              <img
                alt="close icon"
                className="icon close"
                src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23FFFFFF' d='M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z' /%3E%3C/svg%3E"
              />
            </button>
            <div
              className="gift-card-info"
            >
              <p
                className="bold left"
              >
                Gift Card
              </p>
              <p
                className="number"
              >
                - 
                1111111111111111
              </p>
              <br />
              Will use
              <p
                className="bold"
              >
                $100
              </p>
              of your 
              <br />
              <p
                className="bold left"
              >
                $200
              </p>
              balance
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`GiftCards renders correctly with invalid fields 1`] = `
Array [
  <input
    aria-label="Use Gift Card"
    checked={true}
    name="Use Gift Card"
    onChange={[Function]}
    type="checkbox"
  />,
  "Use gift card",
  <br />,
  <p
    className="note"
  >
    Note: Adding or removing gift cards may require you to re-enter other payment information below.
  </p>,
  " ",
  <p
    className="invalid-message"
  >
    *Invalid
     
    test field
  </p>,
  <div
    className="grid-x"
  >
    <div
      className="applied-gift-cards small-12"
    >
      <h4>
        Applied Gift Cards
      </h4>
      <div
        className="grid-x"
      >
        <div
          className="small-12 medium-7 large-6 applied-gift-card-container"
        >
          <div
            className="applied-gift-card card"
          >
            <button
              aria-label="Remove Gift Card"
              className="remove-card"
              onClick={[Function]}
              tabIndex="0"
              value="Remove Gift Card"
            >
              <img
                alt="close icon"
                className="icon close"
                src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23FFFFFF' d='M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z' /%3E%3C/svg%3E"
              />
            </button>
            <div
              className="gift-card-info"
            >
              <p
                className="bold left"
              >
                Gift Card
              </p>
              <p
                className="number"
              >
                - 
                1111111111111111
              </p>
              <br />
              Will use
              <p
                className="bold"
              >
                $100
              </p>
              of your 
              <br />
              <p
                className="bold left"
              >
                $200
              </p>
              balance
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
